### PR TITLE
debug: Add set_pos function

### PR DESCRIPTION
**Pull request by dracc**
Expose a function in `debug.nut` for teleporting to any coordinate.

### CODE REVIEW

File: scripts/module/debug.nut [hunk 1]
```diff
@@ -12,8 +12,19 @@ function onPlayerCommand( player, cmd, text ) {
...
+        local pos = Vector(coords[0].tointeger(), coords[1].tointeger(), coords[2].tointeger());
+        TeleportPlayer(player, pos, 0);
+        MessagePlayer("Teleported to (" + text + ").", player);
+      } else {
+        MessagePlayer("[#ff0000]Usage: /" + cmd + " X Y Z", player);
+      }
+    }
   } else if (cmd == "help") {
-    MessagePlayer("[#8080A0]/r[eload] /pos", player);
+    MessagePlayer("[#8080A0]/r[eload] /pos /g[o]t[o]p[os]", player);
```
Comment (JayFoxRox) [inline scripts/module/debug.nut]: yikes.. this is ugly. How about `set_pos` and `get_pos` instead?

Generally: You could also extend the `/t[eleport]` command to accept coordinates (and possibly restrict that feature to admins)

If we implement #15 we also don't have to worry about long command names anymore anyway

File: scripts/module/debug.nut [hunk 2]
```diff
@@ -12,8 +12,19 @@ function onPlayerCommand( player, cmd, text ) {
...
     print(player.Pos);
+  } else if (cmd == "gtp" || cmd == "gotopos") {
+    if (text) {
+      local coords = split(text, " ");
+      if (IsNum(coords[0]) && IsNum(coords[1]) && IsNum(coords[2])) {
+        local pos = Vector(coords[0].tointeger(), coords[1].tointeger(), coords[2].tointeger());
+        TeleportPlayer(player, pos, 0);
+        MessagePlayer("Teleported to (" + text + ").", player);
+      } else {
+        MessagePlayer("[#ff0000]Usage: /" + cmd + " X Y Z", player);
```
Comment (JayFoxRox) [inline scripts/module/debug.nut]: For other stuff you wrote `<X> <Y> <Z>` - consistency?

**Comment by dracc** [comment]
`IsNum()` check has been removed as it can't handle floats.
`.tointeger()` has been replaced with `.tofloat()` for that sweet sweet precision.

**Comment by JayFoxRox** [comment]
Did you consider integrating this command with `teleporter.nut` (as I suggested above)?

Anyway: Please squash up, this looks mergeable as-is and we can optionally refactor later.

File: scripts/module/teleporter.nut [hunk 3]
```diff
@@ -21,11 +21,16 @@ function teleport(player, location) {
...
 function onPlayerCommand( player, cmd, text ) {
   cmd = cmd.tolower();
   if(cmd == "t" || cmd == "teleport") {
     if(text) {
+      local coords = split(text, " ");
+      if (coords.len == 3 && player.Admin) {
+        local pos = Vector(coords[0].tofloat(), coords[1].tofloat(), coords[2].tofloat());
+        TeleportPlayer(player, pos, 0);
+        return 1;
+      }
```
Comment (JayFoxRox) [inline scripts/module/teleporter.nut]: nit: I'd surround this block with empty lines to make sure that people can see it's independent

File: scripts/module/debug.nut [hunk 4]
```diff
@@ -10,10 +10,10 @@ function onPlayerCommand( player, cmd, text ) {
 
     collectgarbage();
     ReloadScripts();
-  } else if (cmd == "pos") {
+  } else if (cmd == "get_pos") {
     print(player.Pos);
   } else if (cmd == "help") {
-    MessagePlayer("[#8080A0]/r[eload] /pos", player);
+    MessagePlayer("[#8080A0]/r[eload] /get_pos", player);
```
Comment (JayFoxRox) [inline scripts/module/debug.nut]: I wouldn't bother with renaming this one, as there's no analogous `/set_pos` anymore.

File: scripts/module/teleporter.nut (hunk 3 above)
Comment (dracc) [reply scripts/module/teleporter.nut]: Better? 🙂

**Pull request merged**